### ISSUE TITLE
slack alerting log not enough context

### ISSUE THREAD

**Issue by melchiormoulin** [prometheus/alertmanager#2176]
**What did you do?**
```
When a slack channel is archived the log do not precise the channel archived.
```
**What did you expect to see?**
```
I expected to see the channel which is archived.
```
**What did you see instead? Under which circumstances?**
```
2020-02-04T09:39:54.511159+00:00 myalertmanagerhost  [7c853103466b11eabe0fd8c497b517b8] level=error ts=2020-02-04T09:39:54.510Z caller=notify.go:367 component=dispatcher msg="Error on notify" err="cancelling notify retry for \"slack\" due to unrecoverable error: unexpected status code 410: \"channel_is_archived\""
```
**Environment**

* Alertmanager version:
0.18.0

* Logs:
```
2020-02-04T09:39:54.511159+00:00 myalertmanagerhost  [7c853103466b11eabe0fd8c497b517b8] level=error ts=2020-02-04T09:39:54.510Z caller=notify.go:367 component=dispatcher msg="Error on notify" err="cancelling notify retry for \"slack\" due to unrecoverable error: unexpected status code 410: \"channel_is_archived\""
```

Hello,
Do you prefer to log it in the slack.go in the Notify method ? 
An other more generic could be in util.go Check method and pass the req var in param of this method to be able to log the request in case of failure?
I can do it myself.

**Comment by simonpasquier** [comment]
I'd rather see the change limited to the Slack notifier. Logging request parameters could leak secret information so we need to be careful.

**Comment by simonpasquier** [comment]
Closed by #2177
